### PR TITLE
core/types: add data gas fields in Receipt

### PR DESCRIPTION
**Pull request by fjl**
This pulls in receipt changes from the 4844-devnet-7 branch.

### CODE REVIEW

File: core/types/receipt.go [hunk 1]
```diff
@@ -324,9 +326,14 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 		// The transaction type and hash can be retrieved from the transaction itself
 		rs[i].Type = txs[i].Type()
 		rs[i].TxHash = txs[i].Hash()
-
 		rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
 
+		// EIP-4844 blob transaction fields
+		if len(txs[i].BlobHashes()) != 0 {
+			rs[i].DataGasUsed = uint64(len(txs[i].BlobHashes()) * params.BlobTxDataGasPerBlob)
```
Comment (karalabe) [inline core/types/receipt.go:333]: `txs[i].BlobGas()`

File: core/types/receipt.go [hunk 2]
```diff
@@ -63,6 +63,8 @@ type Receipt struct {
 	ContractAddress   common.Address `json:"contractAddress"`
 	GasUsed           uint64         `json:"gasUsed" gencodec:"required"`
 	EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"` // required, but tag omitted for backwards compatibility
+	DataGasUsed       uint64         `json:"dataGasUsed,omitempty"`
+	DataGasPrice      *big.Int       `json:"dataGasPrice,omitempty"`
```
Comment (fjl) [inline core/types/receipt.go:67]: @karalabe says this should be called `BlobGasUsed`

**Review by rjl493456442 — APPROVED** [review]
Except the field name, lgtm

**Review by karalabe — APPROVED** [review]
SGTM, I'll fix all the names in a followup PR

**Pull request merged**